From a9ff85d6515a3e1dbadc617d9e8b84fb1db3cb70 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 16 Oct 2024 23:15:49 +0200
Subject: [PATCH] Use AdwStyleManager

Uses Adw's new style manager to get the accent color.
---
 graphs/canvas.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/graphs/canvas.py b/graphs/canvas.py
index 4dffbb9c..8d7d4b35 100644
--- a/graphs/canvas.py
+++ b/graphs/canvas.py
@@ -11,7 +11,7 @@ interactive navigation in conjunction with graphs-specific structures.
 """
 from contextlib import nullcontext
 
-from gi.repository import GObject, Gdk, Graphs, Gtk
+from gi.repository import Adw, GObject, Gdk, Graphs, Gtk
 
 from graphs import artist, misc, scales, utilities
 
@@ -172,7 +172,8 @@ class Canvas(Graphs.Canvas, FigureCanvas):
         def rgba_to_tuple(rgba):
             return (rgba.red, rgba.green, rgba.blue, rgba.alpha)
 
-        rgba = self.get_style_context().lookup_color("accent_color")[1]
+        style_manager = Adw.StyleManager()
+        rgba = style_manager.get_accent_color_rgba()
         self.rubberband_edge_color = rgba_to_tuple(rgba)
         rgba.alpha = 0.3
         self.rubberband_fill_color = rgba_to_tuple(rgba)
-- 
2.47.0

